scripts/release/cut-release-branch.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+: '
+So you are looking to cut a release branch? Well you came
+to the right script.
+
+This script can be used to cut any branch on any repository
+
+For `pytorch/executorch` usage would be like:
+> DRY_RUN=disabled cut-release-branch.sh
+
+or to cut from main branch:
+> DRY_RUN=disabled GIT_BRANCH_TO_CUT_FROM=main cut-release-branch.sh
+'
+
+set -eou pipefail
+
+GIT_TOP_DIR=$(git rev-parse --show-toplevel)
+GIT_REMOTE=${GIT_REMOTE:-origin}
+GIT_BRANCH_TO_CUT_FROM=${GIT_BRANCH_TO_CUT_FROM:-viable/strict}
+
+# should output something like 1.11
+RELEASE_VERSION=${RELEASE_VERSION:-$(cut -d'.' -f1-2 "${GIT_TOP_DIR}/version.txt")}
```

Comment:
Do we want to name the release branch with the full version or just the MAJOR.MINOR version?

Reply:
In PyTorch, the release branch name is just `release/MAJOR.MINOR`, for example `release/2.2`, which is corresponding to the major release `2.2.0`.  Then, we have patch release, i.e. `2.2.1`, 2.2.2` after that to fix bugs that people report on that release branch.  The timeline for a major PyTorch release is every 3 month, so there are 2 patch release in between before the next major release.  Only major release requires doing branch cut, `release/2.3` is next.